lib/cocoapods/sources_manager.rb

```diff
@@ -208,11 +253,14 @@ def update(source_name = nil, show_output = false)
           sources =  git_sources
         end
 
+        changed_spec_paths = {}
         sources.each do |source|
           UI.section "Updating spec repo `#{source.name}`" do
             Dir.chdir(source.repo) do
               begin
+                prev_commit_hash = (git! %w(rev-parse HEAD)).strip
```

Comment:
can we move this updating logic into `Source` ? i.e. so it has an `update` method that returns the changes spec paths?